semanticdb/metacp/src/main/scala/scala/meta/internal/scalacp/TypeOps.scala

```diff
@@ -60,95 +48,92 @@ trait TypeOps { self: Scalacp =>
-                case true => s.SingletonType(st.BOOLEAN, None, Symbols.None, 1, "")
-                case x: Byte => s.SingletonType(st.BYTE, None, Symbols.None, x.toLong, "")
-                case x: Short => s.SingletonType(st.SHORT, None, Symbols.None, x.toLong, "")
-                case x: Char => s.SingletonType(st.CHAR, None, Symbols.None, x.toLong, "")
-                case x: Int => s.SingletonType(st.INT, None, Symbols.None, x.toLong, "")
-                case x: Long => s.SingletonType(st.LONG, None, Symbols.None, x, "")
-                case x: Float => s.SingletonType(st.FLOAT, None, Symbols.None, floatBits(x), "")
-                case x: Double => s.SingletonType(st.DOUBLE, None, Symbols.None, doubleBits(x), "")
-                case x: String => s.SingletonType(st.STRING, None, Symbols.None, 0, x)
-                case null => s.SingletonType(st.NULL, None, Symbols.None, 0, "")
-                case other => sys.error(s"unsupported const $other")
-              }
+            def floatBits(x: Float) = java.lang.Float.floatToRawIntBits(x).toLong
+            def doubleBits(x: Double) = java.lang.Double.doubleToRawLongBits(x)
+            const match {
+              case () =>
+                s.SingletonType(st.UNIT, s.NoType, Symbols.None, 0, "")
+              case false =>
+                s.SingletonType(st.BOOLEAN, s.NoType, Symbols.None, 0, "")
+              case true =>
+                s.SingletonType(st.BOOLEAN, s.NoType, Symbols.None, 1, "")
+              case x: Byte =>
+                s.SingletonType(st.BYTE, s.NoType, Symbols.None, x.toLong, "")
+              case x: Short =>
+                s.SingletonType(st.SHORT, s.NoType, Symbols.None, x.toLong, "")
+              case x: Char =>
+                s.SingletonType(st.CHAR, s.NoType, Symbols.None, x.toLong, "")
+              case x: Int =>
+                s.SingletonType(st.INT, s.NoType, Symbols.None, x.toLong, "")
+              case x: Long =>
+                s.SingletonType(st.LONG, s.NoType, Symbols.None, x, "")
+              case x: Float =>
+                s.SingletonType(st.FLOAT, s.NoType, Symbols.None, floatBits(x), "")
+              case x: Double =>
+                s.SingletonType(st.DOUBLE, s.NoType, Symbols.None, doubleBits(x), "")
+              case x: String =>
+                s.SingletonType(st.STRING, s.NoType, Symbols.None, 0, x)
+              case null =>
+                s.SingletonType(st.NULL, s.NoType, Symbols.None, 0, "")
+              case other =>
+                sys.error(s"unsupported const $other")
             }
-            Some(s.Type(tag = stag, singletonType = Some(stpe)))
           case RefinedType(sym, parents) =>
-            val stag = t.STRUCTURAL_TYPE
-            val stpe = {
-              val sparents = parents.flatMap(loop)
-              Some(s.Type(tag = t.WITH_TYPE, withType = Some(s.WithType(sparents))))
-            }
+            val sparents = parents.map(loop).filter(_.isDefined)
```

Comment:
Also, can we have `isEmpty` and `nonEmpty` on s.Type?

Reply:
I added an extension method `nonEmpty` which aliases to `isDefined`